### PR TITLE
Sync request() and ws_connect()

### PR DESCRIPTION
**Pull request by asvetlov**
Fix for #2292

### CODE REVIEW

**Comment by asvetlov** [comment]
Any review?
If not -- I'll merge it tomorrow evening.

**Comment by cecton** [comment]
Actually I don't check every day that there is new PR so it's a good thing you added me. I'm reviewing now

File: docs/client_reference.rst [hunk 1]
```diff
@@ -524,29 +527,41 @@ The client session supports the context manager protocol for self closing.
...
-         Add :meth:`ws_connect`.
+      :param bytes fingerprint: Pass the SHA256 digest of the expected
+           certificate in DER format to verify that the certificate the
+           server presents matches. Useful for `certificate pinning
+           <https://en.wikipedia.org/wiki/Transport_Layer_Security#Certificate_pinning>`_.
 
-      .. versionadded:: 0.18
+           Note: use of MD5 or SHA1 digests is insecure and deprecated.
 
-         Add *auth* parameter.
```
Comment (cecton) [inline docs/client_reference.rst:537]: Did you intentionally removed part of the documentation?

Comment (asvetlov) [reply docs/client_reference.rst:537]: Yes.
I've dropped `versioadded` records for versions < 2.0
They don't carry any useful information now.

Comment (cecton) [reply docs/client_reference.rst:537]: Yup it was quite noisy after all.

**Comment by cecton** [comment]
You could also rebase instead of merging the master branch to the feature branch, it will prevent having merge commits in the tree that doesn't carry any useful information.

**Comment by cecton** [comment]
Ahhh I see! That's why you use the squash and merge button, it removes the merge commit at the same time.

**Comment by asvetlov** [comment]
Yes, with squashing I don't care about commits history

**Pull request merged**